api/v1alpha6/conversion.go

```diff
@@ -60,14 +60,11 @@ func restorev1alpha7MachineSpec(previous *infrav1.OpenStackMachineSpec, dst *inf
 	// PropagateUplinkStatus has been added in v1alpha7.
 	// We restore the whole Ports since they are anyway immutable.
 	dst.Ports = previous.Ports
+	dst.DataVolumes = previous.DataVolumes
 }
 
 func restorev1alpha7Bastion(previous **infrav1.Bastion, dst **infrav1.Bastion) {
-	// PropagateUplinkStatus has been added in v1alpha7.
-	// We restore the whole Ports since they are anyway immutable.
-	if *previous != nil && (*previous).Instance.Ports != nil && *dst != nil && (*dst).Instance.Ports != nil {
-		(*dst).Instance.Ports = (*previous).Instance.Ports
-	}
+	restorev1alpha7MachineSpec(&(*previous).Instance, &(*dst).Instance)
```

Comment:
Doesn't this panic if (*previous) is null? There maybe some context-dependent reason why that never ends up happening, but to make me happy could we keep the previous null checks before dereferencing (*previous) and (*dst) to call restorev1alpha7MachineSpec?

Reply:
👍